### PR TITLE
NIFI-7969: ValidateRecord enhanced with Force Types From Schema property

### PR DESCRIPTION
**Pull request by pgyori**
https://issues.apache.org/jira/browse/NIFI-7969

#### Description of PR

This commit adds Force Types From Schema property to the ValidateRecord processor and contains documentation update for the processor.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateRecord.java [hunk 1]
```diff
@@ -180,6 +180,19 @@
         .defaultValue("true")
         .required(true)
         .build();
+    static final PropertyDescriptor COERCE_TYPES = new PropertyDescriptor.Builder()
+            .name("coerce-types")
+            .displayName("Force Types From Schema")
+            .description("If enabled, the processor will coerce every field to the type specified in the schema. "
```
Comment (turcsanyip) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateRecord.java:186]: I would mention in the description (and maybe also in the property name) that the schema here is the Record Reader's schema, not the validation schema (which may be a different one).

Comment (pgyori) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateRecord.java]: Thank you! I pushed a new commit with the clarification.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/resources/docs/org.apache.nifi.processors.standard.ValidateRecord/additionalDetails.html [hunk 2]
```diff
@@ -29,13 +29,23 @@
...
 
 <p>
-    The value of the Force Types From Schema property has an effect on what
+    The processor first reads the data from the incoming FlowFile using the specified Record Reader,
+    which uses a schema. Then, depending on the value of the Schema Access Strategy property, the processor
+    can either use the reader's schema, or a different schema to validate the data against.
+    After that, the processor writes the data into the outgoing FlowFile using the specified
+    Record Writer. If the data is valid, the validation schema is used by the writer.
+    If the data is invalid, the writer uses the reader's schema.
+    The  <b>Force Types From Schema</b> property affects the first step:
```
[the remark this answers is not in the document]
Comment (pgyori) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/resources/docs/org.apache.nifi.processors.standard.ValidateRecord/additionalDetails.html]: Thank you! The fix is in my new commit.

**Review by turcsanyip — APPROVED** [review]
+1 LGTM
Merging to main.

**Pull request closed without merging**